src/spikeinterface/generation/_template_database.py

```diff
@@ -0,0 +1,47 @@
+from spikeinterface.core.template import Templates
+from spikeinterface.core import generate_sorting, InjectTemplatesRecording
+import zarr
+
+
+def fetch_templates_from_database(dataset="test_templates"):
+
+    import s3fs
+
+    s3 = s3fs.S3FileSystem(anon=False, client_kwargs={"region_name": "us-east-2"})
+
+    # Specify the S3 bucket and path where your Zarr dataset is stored
+    store = s3fs.S3Map(root=f"spikeinterface-template-database/{dataset}", s3=s3)
+
+    # Load the Zarr group from S3
+    zarr_group = zarr.open(store, mode="r")
+
+    templates_object = Templates.from_zarr_group(zarr_group)
+
+    return templates_object
+
+
+def generate_recording_from_template_database(selected_unit_inidces=None, dataset="test_templates", durations=None):
```

Comment:
For we should be coordinate with a call.

I started a drift generator with synthetic templates here : https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/generation/drifting_generator.py#L249

In my mind this function should be extended or refactored to accept real templates then the sorting code and the drift code could be shared.
Then this function could be a simple wrapper on top on the other function that would download template and call the other one.

Reply:
Sure, maybe tomorrow after our meeting?